layers/layer_options.cpp

```diff
@@ -381,51 +432,73 @@ void ProcessConfigAndEnvSettings(ConfigAndEnvSettings *settings_data) {
+    // Process layer enable settings
+    std::string cfg_enables = getLayerOption((layer_ns + ".enables").c_str());
+    std::string env_enables = GetEnvironment("VK_LAYER_ENABLES");
+    SetLocalEnableSetting(cfg_enables, ",", settings_data->enables);
+    SetLocalEnableSetting(env_enables, env_delimiter, settings_data->enables);
+
+    SetValidationSetting(settings_data->enables, best_practices, "validate_best_practices");
+    SetValidationSetting(settings_data->enables, vendor_specific_arm, "validate_best_practices_arm");
+    SetValidationSetting(settings_data->enables, vendor_specific_amd, "validate_best_practices_amd");
+    SetValidationSetting(settings_data->enables, vendor_specific_img, "validate_best_practices_img");
+    SetValidationSetting(settings_data->enables, vendor_specific_nvidia, "validate_best_practices_nvidia");
+    SetValidationSetting(settings_data->enables, sync_validation, "validate_sync");
+    SetValidationSetting(settings_data->enables, sync_validation_queue_submit, "sync_queue_submit");
+    SetValidationGPUBasedSetting(settings_data->enables, "validate_gpu_based");
+    SetValidationSetting(settings_data->enables, gpu_validation_reserve_binding_slot, "reserve_binding_slot");
+
+    // Process layer disable settings
+    std::string cfg_disables = getLayerOption((layer_ns + ".disables").c_str());
+    std::string env_disables = GetEnvironment("VK_LAYER_DISABLES");
+    SetLocalDisableSetting(cfg_disables, ",", settings_data->disables);
+    SetLocalDisableSetting(env_disables, env_delimiter, settings_data->disables);
+
+    SetValidationSetting(settings_data->disables, stateless_checks, "stateless_param");
+    SetValidationSetting(settings_data->disables, thread_safety, "thread_safety");
+    SetValidationSetting(settings_data->disables, core_checks, "validate_core");
+    SetValidationSetting(settings_data->disables, command_buffer_state, "check_command_buffer");
+    SetValidationSetting(settings_data->disables, object_in_use, "check_object_in_use");
+    SetValidationSetting(settings_data->disables, query_validation, "check_query");
+    SetValidationSetting(settings_data->disables, image_layout_validation, "check_image_layout");
+    SetValidationSetting(settings_data->disables, handle_wrapping, "unique_handles");
+    SetValidationSetting(settings_data->disables, object_tracking, "object_lifetime");
+    SetValidationSetting(settings_data->disables, shader_validation, "check_shaders");
+    SetValidationSetting(settings_data->disables, shader_validation_caching, "check_shaders_caching");
+
     // Process message filter ID list
-    CreateFilterMessageIdList(list_of_config_filter_ids, ",", settings_data->message_filter_list);
-    CreateFilterMessageIdList(list_of_env_filter_ids, env_delimiter, settings_data->message_filter_list);
+    std::string cfg_filter_ids = getLayerOption((layer_ns + ".message_id_filter").c_str());
+    std::string env_filter_ids = GetEnvironment("VK_LAYER_MESSAGE_ID_FILTER");
```

Comment:
const correctness issues in this function.

```suggestion
    const std::string cfg_filter_ids = getLayerOption((layer_ns + ".message_id_filter").c_str());
    const std::string env_filter_ids = GetEnvironment("VK_LAYER_MESSAGE_ID_FILTER");
```